### PR TITLE
Update for core 2.4 release

### PR DESCRIPTION
**Pull request by bencefr**
- updated engine dependency
- updated serialport handling
- new firmware provided for upcoming device
- new device selector actions and device setup configuration

### CODE REVIEW

File: index.jsx [hunk 1]
```diff
@@ -119,26 +120,45 @@ export default {
...
+        switch (action.type) {
+            case 'DEVICE_DESELECTED':
+                logger.info('Deselecting device');
+                store.dispatch(RssiActions.close());
+                break;
+
+            case 'DEVICE_SETUP_COMPLETE':
+                logger.info(`Selecting device with s/n ${action.device.serialNumber}`);
+                store.dispatch(RssiActions.open(action.device.serialport));
+                break;
```
Comment (mrodem) [inline index.jsx]: Could also listen for `DEVICE_SELECTED` and log something like "Validating formware for device with s/n".

File: index.jsx [hunk 2]
```diff
@@ -119,26 +120,45 @@ export default {
...
-            }));
+
+        switch (action.type) {
+            case 'DEVICE_DESELECTED':
+                logger.info('Deselecting device');
+                store.dispatch(RssiActions.close());
+                break;
+
+            case 'DEVICE_SETUP_COMPLETE':
+                logger.info(`Selecting device with s/n ${action.device.serialNumber}`);
```
Comment (mrodem) [inline index.jsx]: Maybe instead of "Selecting device" here, it should be "Opening device"?

File: actions/rssiActions.js [hunk 3]
```diff
@@ -152,7 +156,8 @@ export function open(serialPort) {
                     throttleUpdates = false;
                     dispatch(rssiData());
                 });
-            });
+            })
+                .on('error', console.log);
```
Comment (mrodem) [inline actions/rssiActions.js]: How about `logger.error` instead?

File: package.json [hunk 4]
```diff
@@ -39,6 +39,6 @@
   },
   "dependencies": {},
   "engines": {
-    "nrfconnect": "2.x"
+    "nrfconnect": "2.4.x"
```
Comment (mrodem) [inline package.json]: Should it be `^2.4.0`? I think it is safe to assume that the app will be supported on v2.4 and any minor version above. We are using semver, so no breaking changes should be introduced in core before v3.0.

File: actions/rssiActions.js (hunk 3 above)
Comment (bencefr) [reply actions/rssiActions.js]: It was on purpose on console.log, I can remove it though.

**Pull request merged**